contrib/golang/common/go/api/filter.go

```diff
@@ -210,3 +210,26 @@ type FilterState interface {
 	SetString(key, value string, stateType StateType, lifeSpan LifeSpan, streamSharing StreamSharing)
 	GetString(key string) string
 }
+
+type MetricType uint32
+
+const (
+	Counter   MetricType = 0
+	Gauge     MetricType = 1
+	Histogram MetricType = 2
+)
+
+type ConfigCallbacks interface {
+	// Define a metric, for different MetricType, name must be different,
+	// for same MetricType, the same name will share a metric.
+	DefineMetric(metricType MetricType, name string) Metric
```

Comment:
also, this could be DefineCounterMetric and DefineGaugeMetric

Reply:
done, thanks.